src/main2.rs

```diff
@@ -0,0 +1,53 @@
+use cpal::{traits::{DeviceTrait, HostTrait, StreamTrait}, StreamConfig};
+use rustfft::{num_complex::Complex, FftPlanner};
+
+fn main() {
+    let stream = build_stream();
+    loop {
+    stream.play().expect("stream couldn't play?");
+    }
+}
+
+fn build_stream() -> cpal::Stream {
+    let host = cpal::default_host();
+    let my_dev = host.default_input_device().expect("no default input device found");
+    let config: StreamConfig = my_dev.default_input_config().expect("No default input config").into();
```

Comment:
This is more a matter of personal preference, but you can eliminate the `config: StreamConfig` type annotation by doing:
`let config = StreamConfig::from(my_dev.default_input_config().expect("No default input config"));`

Reply:
Where do you preferences lie wrt type annotations? I'm assuming there's only so much readability it can grant.

Reply:
Generally, type annotations should only be used to help the compiler figure out the types. They should be avoided whenever possible, because they're usually unnecessary, or there's an alternate way to do things.

Why do I suggest avoiding them? Rust is a generics-heavy language, and sometimes types can be complex (or even impossible to explicitly name). It's better to let the compiler do type inference, since if you change the types in your code later on, it won't break compilation.